client/jetpack-onboarding/main.jsx

```diff
@@ -188,8 +199,11 @@ export default connect(
 		const settings = getJetpackOnboardingSettings( state, siteId );
 		const isBusiness = get( settings, 'siteType' ) === 'business';
 
+		// We really want `isRequestingSite( state, siteSlug )` (not `siteId`).
+		// For the rationale, see the comment right above the `<QuerySites />` component
+		// further up in this file.
 		const isRequestingWhetherConnected =
-			isRequestingSite( state, siteId ) || isRequestingSites( state );
+			isRequestingSite( state, siteSlug ) || isRequestingSites( state );
```

Comment:
According to its JSdoc `isRequestingSite` accepts only a numerical `siteId`. The endpoint supports a slug, so it probably makes sense to update it in the selector docs too, as it's a little weird to use a non-documented case like this.

Reply:
#23526